src/Nio/ZynSema.h

```diff
@@ -12,45 +12,38 @@ class ZynSema
     {
     }
 
-    ~ZynSema (void)
-    {
-        pthread_mutex_destroy (&_mutex);
-        pthread_cond_destroy (&_cond);
-    }
-
     int init (int, int v)
     {
         _count = v;
-        return pthread_mutex_init (&_mutex, 0) || pthread_cond_init (&_cond, 0);
+        return 0;
     }
 
     int post (void)
     {
-        pthread_mutex_lock (&_mutex);
-        if (++_count == 1) pthread_cond_signal (&_cond);
-        pthread_mutex_unlock (&_mutex);
+        _mutex.lock();
```

Comment:
Why this function uses manual lock-unlock calls whereas `wait` uses `std::unique_lock`?

Reply:
My aim is not to improve the code by switching to RAII lock guards; I'm only trying to get the existing code to build with MSVC. I chose to keep manual lock-unlock calls as then each new line clearly corresponds to one old line. This should hopefully make resolving future conflicts with any upstream changes easier, as well as make this PR easier to review. However, `std::condition_variable::wait` requires a `std::unique_lock` as an argument, so I had to make an exception for that one function.